rust_src/src/windows.rs

```diff
@@ -713,6 +713,40 @@ pub extern "C" fn decode_any_window(window: LispObject) -> LispWindowRef {
     LispWindowOrSelected::from(window).into()
 }
 
+/// Return the normal height of window WINDOW.
+/// WINDOW must be a valid window and defaults to the selected one.
+/// If HORIZONTAL is non-nil, return the normal width of WINDOW.
+///
+/// The normal height of a frame's root window or a window that is
+/// horizontally combined (a window that has a left or right sibling) is
+/// 1.0.  The normal height of a window that is vertically combined (has a
+/// sibling above or below) is the fraction of the window's height with
+/// respect to its parent.  The sum of the normal heights of all windows in a
+/// vertical combination equals 1.0.
+///
+/// Similarly, the normal width of a frame's root window or a window that is
+/// vertically combined equals 1.0.  The normal width of a window that is
+/// horizontally combined is the fraction of the window's width with respect
+/// to its parent.  The sum of the normal widths of all windows in a
+/// horizontal combination equals 1.0.
+///
+/// The normal sizes of windows are used to restore the proportional sizes
+/// of windows after they have been shrunk to their minimum sizes; for
+/// example when a frame is temporarily made very small and afterwards gets
+/// re-enlarged to its previous size.
+#[lisp_fn(min = "0")]
+pub fn window_normal_size(
+    window: LispWindowValidOrSelected,
+    horizontal: LispObject,
+) -> EmacsDouble {
+    let win: LispWindowRef = window.into();
+    EmacsDouble::from(if horizontal.is_nil() {
+        win.normal_lines
+    } else {
+        win.normal_cols
+    })
+}
```

Comment:
I might have written this in the following style.

```
let value = if horizontal {
win.normal_cols
} else {
win.normal_lines
};
value.into() // or EmacsDouble::from(value)

Reply:
Done